tests/apps/test_similarity.py

```diff
@@ -368,71 +451,230 @@ def test_mean_event(self, monkeypatch):
+        below zero."""
         graph = nx.complete_graph(5)
-        mock_eng_run = mock.MagicMock()
+        with pytest.raises(ValueError, match="Cannot request orbits with photon number below zero"):
+            similarity.feature_vector_orbits(graph, [[-1, 1]])
+
+    def test_invalid_n_mean(self):
+        """Test if function raises a ``ValueError`` when the mean photon number is specified to
+        be negative."""
+        g = nx.complete_graph(5)
+        with pytest.raises(ValueError, match="Mean photon number must be non-negative"):
+            similarity.feature_vector_orbits(g, [[1, 1], [2]], n_mean=-1)
+
+    def test_invalid_loss(self):
+        """Test if function raises a ``ValueError`` when the loss parameter is specified outside
+        of range."""
+        g = nx.complete_graph(5)
+        with pytest.raises(
+            ValueError, match="Loss parameter must take a value between zero and one"
+        ):
+            similarity.feature_vector_orbits(g, [[1, 1], [2]], loss=2)
+
+    def test_calls_exact_for_zero_samples(self):
+        """Test if function calls the exact function for zero samples"""
+        g = nx.complete_graph(5)
+        assert similarity.feature_vector_orbits(
+            g, [[1, 1]], samples=0
+        ) == similarity.prob_orbit_exact(g, [1, 1])
+
+    def test_correct_vector_returned(self, monkeypatch):
+        """Test if function correctly constructs the feature vector. The ``prob_orbit_exact``
+         and ``prob_orbit_mc`` function called within ``feature_vector_orbits`` are
+         monkeypatched to return hard-coded outputs that depend only on the orbit."""
 
         with monkeypatch.context() as m:
-            m.setattr(sf.LocalEngine, "run", mock_eng_run)
-            similarity.prob_event_mc(graph, 6, 3, samples=1)
-            p_func = mock_eng_run.call_args[0][0]
+            m.setattr(
+                similarity,
+                "prob_orbit_mc",
+                lambda _graph, orbit, n_mean, samples, loss: 1.0 / sum(orbit),
+            )
+            graph = nx.complete_graph(8)
 
-        assert not all([isinstance(op, sf.ops.LossChannel) for op in p_func.circuit])
+            assert similarity.feature_vector_orbits(graph, [[1, 1], [2, 1, 1]], samples=1) == [
+                0.5,
+                0.25,
+            ]
 
-    def test_all_loss(self, monkeypatch):
-        """Test if function samples from the vacuum when maximum loss is applied."""
-        dim = 5
-        graph = nx.complete_graph(dim)
-        mock_eng_run = mock.MagicMock()
+        with monkeypatch.context() as m:
+            m.setattr(
+                similarity,
+                "prob_orbit_exact",
+                lambda _graph, orbit, n_mean, loss: 0.5 * (1.0 / sum(orbit)),
+            )
+            graph = nx.complete_graph(8)
+
+            assert similarity.feature_vector_orbits(graph, [[1, 1], [2, 1, 1]]) == [0.25, 0.125]
+
+
+class TestFeatureVectorEvents:
```

Comment:
Also needs to test that correct values are calculated for non-trivial specific cases